#
# Committed-By: Marc Kleine-Budde <mkl@pengutronix.de>, 2006-11-29
#
# Error:
#
# check for SHA1 is not cross aware, make it cacheable and added cross default value
#
# State:
#
# should be commited
#
Index: pure-ftpd-1.0.21/configure.ac
===================================================================
--- pure-ftpd-1.0.21.orig/configure.ac
+++ pure-ftpd-1.0.21/configure.ac
@@ -1078,7 +1078,9 @@ then
 MD5 functions])
 fi
 
-AC_MSG_CHECKING(whether you already have a standard SHA1 implementation)
+AC_CACHE_CHECK([whether you already have a standard SHA1 implementation],
+  [ac_cv_use_system_crypt_sha1],
+[
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdio.h>
 #include <string.h>
@@ -1097,12 +1099,15 @@ int main(void)
     
     return strcasecmp(b, "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3");
 }
-]])],[
-AC_MSG_RESULT(yes)
-AC_DEFINE(USE_SYSTEM_CRYPT_SHA1,,[Define if you already have standard
+]])],[ac_cv_use_system_crypt_sha1=yes
+],[ac_cv_use_system_crypt_sha1=no
+],[ac_cv_use_system_crypt_sha1="no (cross compiling)"])
+])
+if test "x$ac_cv_use_system_crypt_sha1" = "xyes"
+then
+  AC_DEFINE(USE_SYSTEM_CRYPT_SHA1,,[Define if you already have standard
 SHA1 functions])
-],[AC_MSG_RESULT(no)
-],[])
+fi
 
 AC_MSG_CHECKING([whether we are inside a Virtuozzo virtual host])
 if test -d /proc/vz; then
